pkg/operator/kit/resource.go

```diff
@@ -72,15 +73,45 @@ func CreateCustomResource(resource CustomResource, clientset kubernetes.Interfac
 	}
 	kubeVersion := version.MustParseSemantic(serverVersion.GitVersion)
 
-	if kubeVersion.AtLeast(version.MustParseSemantic("v1.7.0")) {
-		return createCRD(resource)
+	if kubeVersion.AtLeast(version.MustParseSemantic(serverVersionV170)) {
+		// Create and wait for CRD resources
+		apiExtClientset, err := getApiextensionsClientset()
```

Comment:
I understand. I was debating this because it requires me to create a new context struct to carry the two clients.

I also wanted to keep the distinction of creating CRD vs TPR (using core clientset vs apiextension clientset) in the kit package. That way the caller doesnt have to worry about this. Otherwise, the caller will need to get the k8s version and pass the appropriate client.